src/erdstall.ts

```diff
@@ -153,6 +153,11 @@ export interface Trader {
 	acceptTrade(offer: TradeOffer): Promise<TxReceipt>;
 }
 
+export interface StagesGenerator {
+	stages: AsyncGenerator<[StageName, ethers.ContractTransaction], void, void>;
```

Comment:
What's actually the purpose of the `StageName` parameter? Can't we just do
```suggestion
	stages: AsyncGenerator<ethers.ContractTransaction, void, void>;
```
? The caller either calls `client.deposit` or `client.withdraw` etc and knows, what he's calling.

Reply:
@ndzik explained to me that it's about differentiating the different calls during deposits. Wouldn't it then actually make sense to replace `StageName` by just a `string` and then write something more specific into it, like `erc721/${tokenAddr}/{id}/approve` or `erc20/${tokenAddr}/approve` etc? Because if deposit is called on multiple assets, only knowing that it's an `approval` or `deposit` call is ambiguous. Anyway, we can think about this in a later issue.